packages/m2m-gateway/src/api/purposeApiConverter.ts

```diff
@@ -0,0 +1,86 @@
+import { m2mGatewayApi, purposeApi } from "pagopa-interop-api-clients";
+import { Logger } from "pagopa-interop-commons";
+import { ApiError } from "pagopa-interop-models";
+import { assertActivePurposeVersionExists } from "../utils/validators/purposeValidator.js";
+import { purposeNotFound } from "../model/errors.js";
+
+export function toGetPurposesApiQueryParams(
+  params: m2mGatewayApi.GetPurposesQueryParams
+): purposeApi.GetPurposesQueryParams {
+  return {
+    eservicesIds: params.eserviceIds,
+    limit: params.limit,
+    offset: params.offset,
+    consumersIds: [],
+    producersIds: [],
+    states: [],
+    excludeDraft: false,
+    name: undefined,
+  };
+}
+
+export function toM2MGatewayApiPurpose({
+  purpose,
+  logger,
+  throwNotFoundError = false,
+}: {
+  purpose: purposeApi.Purpose;
+  logger: Logger;
+  throwNotFoundError?: boolean;
```

Comment:
This property naming is a bit unclear.
Throw not found error when? Why should this throw this error? 
My proposals:
- `mapThrownErrorsToNotFound`
- `throwNotFoundOnMissingActiveVersion`

Reply:
👌  I'm gonna do the same in #1848 - I like `mapThrownErrorsToNotFound`

Reply:
Done https://github.com/pagopa/interop-be-monorepo/pull/1848/commits/5faadd391ca93a208d244d00d80a70269eb89920